src/ui/components/DevToolsProcessingScreen.tsx

```diff
@@ -45,5 +81,43 @@ export function DevToolsProcessingScreen() {
     message = `Processing... (${Math.round(processingProgress)}%)`;
   }
 
-  return <LoadingScreen message={message} secondaryMessage={secondaryMessage} />;
+  const onChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      const granted = await requestNotificationPermission();
+      if (!granted) {
+        event.target.checked = false;
+        event.target.blur();
```

Comment:
I'm not an a11y expert, so take that with a grain of salt, but I consider `.blur()` to be an anti-pattern. Focus management should be intentional. If we want to change the currently focused element that's fine but to do that we need some kind of a target. IIRC, it's preferred to always have *some* element focused and `.blur()` breaks that rule.

Reply:
I added this initially bc I didn't want focus to be left on a _disabled_ looking input, but since I also disabled the input then this would be unnecessary anyway. At the same time, it's effectively the same thing to my knowledge (the input will be blurred).